dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilderTest.java

```diff
@@ -234,14 +233,34 @@ public void testConfigureAdditionalMavenRepositories_notToUseMavenCentral()
     // This artifact does not exist in Android's repository
     Artifact artifact = new DefaultArtifact("com.google.guava:guava:28.2-jre");
 
-    try {
-      graphBuilder.buildCompleteGraph(new Dependency(artifact, "compile"));
-      fail("The dependency resolution should fail if Maven Central is not used");
-    } catch (DependencyResolutionException ex) {
-      Truth.assertThat(ex.getMessage())
-          .startsWith(
-              "Could not find artifact com.google.guava:guava:jar:28.2-jre in "
-                  + " (https://dl.google.com/dl/android/maven2)");
-    }
+    DependencyGraphResult result =
+        graphBuilder.buildCompleteGraph(new Dependency(artifact, "compile"));
+    Truth.assertThat(result.getArtifactProblems())
+        .comparingElementsUsing(problemOnArtifact)
+        .contains("com.google.guava:guava:28.2-jre");
+  }
+
+  @Test
+  public void testBuildLinkageCheckDependencyGraph_catchRootException() throws RepositoryException {
+    // This should not throw exception
+    DependencyGraphResult result =
+        dependencyGraphBuilder.buildLinkageCheckDependencyGraph(
+            ImmutableList.of(new DefaultArtifact("ant:ant:jar:1.6.2")));
```

Comment:
This is the motivation of this PR. Before this PR, the function threw DependencyResolutionException. Such errors should be reported as `result.getArtifactProblems()`.